controllers/clusterversion_controller.go

```diff
@@ -234,6 +287,28 @@ func (c *ClusterVersionReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 		return ctrl.Result{}, err
 	}
 
+	prometheusRule, err := parsePrometheusRule(pvcPrometheusRules)
+	if err != nil {
+		c.log.Error(err, "Unable to retrieve prometheus rules.", "prometheusRule", klog.KRef(prometheusRule.Namespace, prometheusRule.Name))
+		return ctrl.Result{}, err
+	}
+	prometheusRule.SetNamespace(c.OperatorNamespace)
+	if err := c.own(prometheusRule); err != nil {
+		return ctrl.Result{}, err
+	}
+	operatorConfig, err := c.getOperatorConfig()
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	applyLabels(operatorConfig.Data["OCS_METRICS_LABELS"], &prometheusRule.ObjectMeta)
```

Comment:
Is there a value add of this approach vs not reconciling labels so that users can directly update it?

Reply:
We want users to keep the resources that someone is managing private. as ocs-client-operator is creating the resource it should be the source of truth for it, and no one should be altering it.

Reply:
Sounds good. But, with configmap, we lose label validation that K8s does for us. Maybe you should add that to log failures due to bad labels vs other failures.

Reply:
sorry i didn't get what label validation we will lose here